public/app/features/logs/utils.ts

```diff
@@ -303,6 +309,35 @@ export const copyText = async (text: string, buttonRef: React.MutableRefObject<E
   }
 };
 
+export async function handleOpenLogsContextClick(
+  event: MouseEvent<HTMLElement>,
+  row: LogRowModel,
+  getRowContextQuery: GetRowContextQueryFn | undefined,
+  onOpenContext: (row: LogRowModel) => void
+) {
+  event.stopPropagation();
+  // if ctrl or meta key is pressed, open query in new Explore tab
+  if (getRowContextQuery && (event.nativeEvent.ctrlKey || event.nativeEvent.metaKey || event.nativeEvent.shiftKey)) {
+    const win = window.open('about:blank');
+    // for this request we don't want to use the cached filters from a context provider, but always want to refetch and clear
+    const query = await getRowContextQuery(row, undefined, false);
+    if (query && win) {
+      const url = urlUtil.renderUrl(locationUtil.assureBaseUrl(`${getConfig().appSubUrl}explore`), {
+        left: JSON.stringify({
+          datasource: query.datasource,
+          queries: [query],
+          range: getDefaultTimeRange(),
+        }),
+      });
+      win.location = url;
```

Comment:
Take a look at locationService instead of using window

Reply:
This is copy paste from the current Panel, but definitely, let's update this.

Reply:
Just re-read the code and it's opening a new window with `const win = window.open('about:blank');`, so this doesn't look like a candidate for locationService. Let me know if I got this wrong.